Blackjack/Game/BlackjackGame.m

```diff
@@ -0,0 +1,262 @@
+    return player.state == PLAYING;
+  }];
+  return [self.players filteredArrayUsingPredicate:predicate];
+}
+
+- (NSArray<Player *> *)playersAvailableForNextFound {
+  NSPredicate *predicate = [NSPredicate predicateWithBlock:^BOOL(Player *player, NSDictionary *bindings) {
+    return player.chips >= MINIMUM_ROUND_BET;
+  }];
+  return [self.players filteredArrayUsingPredicate:predicate];
+}
+
+// MARK: - Game Logic
+
+- (void)collectDecision:(Decision)decision {
+  BOOL isLastPlayer = self.currentPlayer == [self currentActivePlayers].lastObject;
+  
+  switch (decision) {
+    case HIT: {
+      [self.currentPlayer acceptNewCard:[self.deck drawRandomCardWithFaceUp:YES]];
+      
+      if (self.currentPlayer.state == PLAYING) {
+        return;
+      }
+      break;
+    }
+      
+    case SURRENDER:
+      [self.currentPlayer setState:BUST];
+      
+    case STAND: {
+      break;
+    }
+  }
+  
+  if (isLastPlayer) {
+    [self finlizeRound];
+  } else {
+    [self nextPlayer];
+  }
+}
+
+- (void)collectBet:(NSInteger)amount {
+  if (self.currentPlayer.chips >= amount) {
+    [self.bets setObject:[NSNumber numberWithInteger:amount] forKey:self.currentPlayer.identifier];
+    [self.currentPlayer collectBet:amount];
+    [self nextPlayer];
+  } else {
+    NSAssert(self.currentPlayer.chips < amount, @"Player should never get to this point");
+  }
+  
+  if (self.bets[self.players.lastObject.identifier]) {
+    [self dealCardsFaceUp:YES];
+    [self dealCardsFaceUp:NO];
+    
+    if (self.currentPlayer == self.players.firstObject && self.currentPlayer.state != PLAYING) {
+      [self nextPlayer];
+    }
+    self.state = BETS_OVER;
+  }
+}
+
+-(void)dealCardsFaceUp:(BOOL)isFaceUp {
+  for (Player *player in self.players) {
+    BOOL _isFaceUp = player == self.players.firstObject ? YES : isFaceUp;
+    [player acceptNewCard:[self.deck drawRandomCardWithFaceUp:_isFaceUp]];
+  }
+  [self.dealer acceptNewCard:[self.deck drawRandomCardWithFaceUp:isFaceUp]];
+}
+
+-(void)nextPlayer {
+  NSMutableArray<NSNumber *> *playersIndexToUpdated = [NSMutableArray new];
+  
+  if (self.currentPlayer != self.players.lastObject) {
+    NSNumber *currentPlayerIndex = [NSNumber numberWithInteger:[self.players indexOfObject: self.currentPlayer]];
+    [playersIndexToUpdated addObject:currentPlayerIndex];
+    
+    Player *nextPlayer = self.players[currentPlayerIndex.intValue + 1];
+    [nextPlayer showHand];
+    self.currentPlayer = nextPlayer;
+    
+    NSNumber *nextPlayerIndex = [NSNumber numberWithInteger:[self.players indexOfObject: nextPlayer]];
+    [playersIndexToUpdated addObject:nextPlayerIndex];
+    
+    [self.delegate focusOnPlayerAtIndex:playersIndexToUpdated.lastObject.integerValue];
+  } else {
+    NSArray<Player *> *currentActivePlayers = [self currentActivePlayers];
+    
+    self.currentPlayer = currentActivePlayers.firstObject;
+    [playersIndexToUpdated addObject:[NSNumber numberWithInteger:0]];
```

Comment:
instead of `[NSNumber numberWithInteger:0]` you can write `@0`.
Do this for all places please